_rules/image-accessible-name-descriptive-qt1vmo.md

```diff
@@ -0,0 +1,295 @@
+rule_type: atomic
+description: |
+  This rule checks that images with an accessible name describe the topic or purpose.
+accessibility_requirements:
+  wcag20:1.1.1: # Non-Text Content (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G94: # Providing short text alternative for non-text content that serves the same purpose and presents the same information as the non-text content
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G95: # Providing short text alternatives that provide a brief description of the non-text content
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgments:
+  authors:
+    - Wilco Fiers
+  assets:
+    - W3C; HTML and W3C logo
+    - Wikimedia; Fireworks image
+    - Adobe; PDF logo
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # (used with `npm test` to ensure validity of code snippets)
+  - 'alt-require'
+  - 'attr-lowercase'
+---
+
+## Applicability
+
+Any `img`, `canvas` or `svg` element that is [visible][] and has a non-empty [accessible name][], except if one of the following is true:
+
+- The element has an [ancestor][] in the [flat tree][] that is [named from author][]; or
+- The element is an `img` element where the [current request][]'s [state][image request state] is not [completely available][].
+
+## Expectation
+
+Each test target has an [accessible name][] that serves an equivalent purpose to the [non-text content][] of that test target.
+
+## Assumptions
+
+This rule assumes that the language of each test target can be correctly determined (either programmatically or by analyzing the content).
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 1.1.1: Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
+- [G94: Providing short text alternative for non-text content that serves the same purpose and presents the same information as the non-text content](https://www.w3.org/WAI/WCAG21/Techniques/general/G94)
+- [G95: Providing short text alternatives that provide a brief description of the non-text content](https://www.w3.org/WAI/WCAG21/Techniques/general/G95)
+- [F30: Failure of Success Criterion 1.1.1 and 1.2.1 due to using text alternatives that are not alternatives (e.g., filenames or placeholder text)](https://www.w3.org/WAI/WCAG21/Techniques/failures/F30)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `img` element has an `alt` attribute that describes the image.
+
+```html
+<html lang="en">
+	<img src="/test-assets/shared/w3c-logo.png" alt="W3C" />
```

Comment:
Wondering if that should be
```suggestion
	<img src="/test-assets/shared/w3c-logo.png" alt="W3C logo" />
```
especially since most other examples seem to have an alt text of "XXX logo".

Reply:
I think either one passes the SC, but generally don't think it should have "logo". That's the difference between describing the image, and describing the **topic or purpose** of the image.

Reply:
But then I don't understand why Passed Example 2 has a description that says "HTML5 logo" and not just "HTML5".

I have no problem with either, but I am a bit puzzled with the fact that in some case the description is "Foo" and in other cases it is "Foo logo". I failed to see the differences prompting this.

Not an important point at all, more curiosity on my side…

Reply:
Fixed the HTML 5 stuff.